test/cli/test_cli_functions.sh

```diff
@@ -284,10 +283,97 @@ function obc_cycle {
     echo_time "✅  obc cycle is done"
 }
 
+function crd_cycle {  
+    crd_array=($(kubectl get crd | awk '{print $1}' | grep -v "NAME"))
+    local crd_create_array=($(test_noobaa silence crd create &>/dev/stdout | awk '{print $7}' | grep -v "Already Exists" | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    local crd_status_array=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | grep -v "Exists" | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    echo_time "Starting the crd cycle" 
+
+    test_noobaa timeout crd create
+    if [ $? -ne 0 ]  
+    then           
+        echo_time "❌ crd doesn't exist in the test. exiting."
+        exit 1 
+     else 
+        echo_time "✅ crd already exsists in the test"   
+    fi  
+  
+    echo_time "checking if crd noobaa create command equals to crd status command:"
+    for crd_status_array in ${crd_array[@]}
+    do
+        if [[ ${crd_create_array[@]} =~ ${crd_status_array} ]]
+        then            
+            echo_time "✅ ${crd_status_array} is exsisted in the crds list"
+        else
+            echo_time "❌ ${crd_status_array} is not existed in the crds list"         
+        fi               
+    done
+         
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅kubectl get crd passed"    
+    fi      
+
+    test_noobaa timeout crd delete 
+    local crd_status_after_delete=($(test_noobaa silence crd status &>/dev/stdout | grep -v "Not Found" | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    echo_time "crds after delete: ${crd_status_after_delete[@]}"
+    
+    for name in ${crd_array[@]} 
+    do
+        if [[ ${crd_status_array[@]} =~ ${crd_status_after_delete} ]]
+        then    
+            echo_time "❌ crd ${name} still exists in the test. exiting."
+            exit 1       
+        else 
+            echo_time "✅ crd ${name} deleted from test" 
+        fi
+    done
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi 
+
+    echo_time "creating crd again for checks..."     
+    test_noobaa silence crd create 
+    
+    for crd_status_array in ${crd_array[@]}
+    do
+        if [[ ${crd_create_array[@]} =~ ${crd_status_array} ]]
```

Comment:
those arrays are populated at the beginning of the function, We need to repopulate.